src/executer.ts

```diff
@@ -0,0 +1,59 @@
+import * as vscode from "vscode";
+import { runSync } from "./run_cmd";
+import { outputWindow } from "./console";
+
+/**
+ * Get the selected text
+ * @returns The selected text
+ */
+function getSelectedText(): string | null {
+    // Get the current editor
+    const editor = vscode.window.activeTextEditor;
+    if (!editor) {
+        return null;
+    }
+
+    const selectedText = editor?.document.getText(editor.selection);
```

Comment:
Here, we can safely assume that `editor` will always have a value since we check for `null` in advance. In js/ts the `var?.var?` operator is called the Optional Chaining Operator and it is used to safely access properties or call methods on an object that might be `null` or `undefined`.